src/controller/handler/connections.ts

```diff
@@ -37,38 +36,75 @@ export class Connections implements IAgentResponseHandler {
         Route will be "topic"
         topic will be "connections"
     */
-    public async handlePost(agentUrl: string, adminApiKey: string, route: string, topic: string, body: any): Promise<any> {
+    public async handlePost(agentUrl: string, agentId: string, adminApiKey: string, route: string, topic: string, body: any): Promise<any> {
+        const delay = (ms: number) => {
+            return new Promise(resolve => setTimeout(resolve, ms));
+        };
+        const readPermission = (governanceKey: string, cacheKey: string) => {
+            this.agentGovernance.readPermission('connections', governanceKey);
+            this.cache.set(cacheKey, {});
+        }
+
         if (route !== 'topic' || topic !== 'connections') {
             throw new ProtocolException('Connections',`${route}/${topic} is not valid.`);
         }
 
-        if (body.state !== 'request') {
-            Logger.info(`${body.state} not applicable`);
-            return;
+        const governanceKey = `${body.state}-${body.initiator}`;
+        const cacheKey = `${agentId}-${governanceKey}`;
+        const permissionState = this.agentGovernance.peekPermission('connections', governanceKey);
+
+        if (AgentGovernance.PERMISSION_DENY === permissionState) {
+             throw new ProtocolException('AgencyGovernance',`${governanceKey} governance doesnt not allow.`);
         }
 
-        if (AgentGovernance.PERMISSION_DENY === this.agentGovernance.getPermission("connections", "accept-invitation")) {
-            throw new ProtocolException('AgencyGovernance',`${topic} governance doesnt not allow.`);
+        // if the cacheKey is in the cache then the agent has already accepted the request
+        // when we only allow once, there is no need to continue with this message
+        if (this.cache.get<any>(cacheKey) && permissionState === AgentGovernance.PERMISSION_ONCE) {
+             return;
         }
 
-        // if the agentUrl is in the cache then the agent has already accepted the request
-        if (this.cache.get<any>(agentUrl)) {
-            return;
+        // this webhook message indicates an agent received an connection
+        // invitation and we want to tell them to accept it, if the policy allows
+        if (body.state === 'invitation' && body.initiator === 'external') {
+            readPermission(governanceKey, cacheKey);
+            await delay(2000);
```

Comment:
I added these delays just because of the uncertainty of of the actual state of the agent and thought that pushing it through immediately could cause a problem.  I have no concrete behavior documented that its needed.  thoughts?